frontend/packages/topology/src/components/dropdowns/ApplicationDropdown.tsx

```diff
@@ -40,10 +45,13 @@ interface ApplicationDropdownProps {
 const ApplicationDropdown: React.FC<ApplicationDropdownProps> = ({ namespace, ...props }) => {
   const { t } = useTranslation();
   const modelFactories = useExtensions<TopologyDataModelFactory>(isTopologyDataModelFactory);
+  const [dynamicModelFactories] = useResolvedExtensions<DynamicTopologyDataModelFactory>(
+    isDynamicTopologyDataModelFactory,
+  );
```

Comment:
Can we update the static extension to make use of `CodeRef` then simplified the resolution with `useResolvedExtensions` 

We probably don't want to render the firehose component until all extensions have been resolved.